src/pages/AddEditPollPage.js

```diff
@@ -0,0 +1,269 @@
+import { useLocation, useNavigate, useParams } from "react-router-dom";
+import Modal from "../Components/Modal";
+
+import {
+  addOption,
+  deleteOption,
+  updateOption,
+} from "../redux/reducers/optionReducer";
+import { MdDelete } from "react-icons/md";
+import { FaEdit } from "react-icons/fa";
+
+const AddEditPoll = () => {
+  const { id } = useParams();
+  const { state } = useLocation();
+  const [newPollData, setNewPollData] = useState({
+    title: "",
+    optionTitle: "",
+  });
+  const [options, setOptions] = useState([]);
+  const [errors, setErrors] = useState({ title: "", optionTitle: "" });
+  const [showModal, setShowModal] = useState(false);
+  const [showDeletedModal, setShowDeletedModal] = useState(false);
+  const [selectedIndex, setSelectedIndex] = useState(null);
+  const [editOption, setEditOption] = useState(null);
+  const { loading } = useSelector((state) => state.pollList);
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  const getSinglePollDetails = async () => {
+    if (state) {
+      setNewPollData({ ...newPollData, title: state.title });
+      setOptions(state.optionList);
+    } else {
+      const result = await dispatch(getSinglePoll(id));
+      if (result?.payload?.status === 200) {
+        setNewPollData({ ...newPollData, title: result?.payload?.data?.title });
+        setOptions(result?.payload?.data?.optionList);
+      }
+    }
+  };
+  useEffect(() => {
+    if (id) {
+      getSinglePollDetails();
+    } else {
+      setNewPollData({ title: "", optionTitle: "" });
+      setOptions([]);
+    }
+  }, [id]);
+
+  const handleOptionChange = (e) => {
+    setNewPollData({ ...newPollData, [e.target.name]: e.target.value });
+    setErrors({ ...errors, [e.target.name]: "" });
+  };
+
+  const handleAddOption = async () => {
+    const { newErrors, isVallid } = validateAddEditForm({
+      optionTitle: newPollData.optionTitle,
+    });
+    if (isVallid) {
+      setOptions([...options, { optionTitle: newPollData.optionTitle }]);
+      if (editOption) {
+        const newOptions = [...options];
+        newOptions[editOption.index].optionTitle = newPollData.optionTitle;
+        setOptions(newOptions);
+      }
+
+      if (id && editOption?.id) {
+        dispatch(
+          updateOption({
+            id: editOption.id,
+            editedOption: newPollData.optionTitle,
+          })
+        );
+      }
+
+      if (id && !editOption?.id) {
+        const result = await dispatch(
+          addOption({
+            id,
+            optionTitle: newPollData.optionTitle,
+          })
+        );
+        if (result?.payload?.status === 200) {
+          const newOptions = [...options, result?.payload?.data.option];
+          setOptions(newOptions);
+        }
+      }
+      setEditOption(null);
+      setNewPollData({ ...newPollData, optionTitle: "" });
+    } else {
+      setErrors(newErrors);
+    }
+  };
+
+  const handleDeleteOption = () => {
```

Comment:
what is handle handle handle every where? addPoll , deletePoll function name not enough? @vikash-kanauja